pkgs/development/python-modules/pex/default.nix

```diff
@@ -2,25 +2,56 @@
   lib,
   buildPythonPackage,
   fetchPypi,
-  hatchling,
-  pythonOlder,
+  fetchurl,
+  setuptools,
 }:
+let
+  # 4 binaries which require vendoring, as otherwise
+  # the build system behind pex will attempt to fetch
+  # them during at build time
+  uv-trampoline = {
+    aarch64-gui = fetchurl {
+      url = "https://raw.githubusercontent.com/astral-sh/uv/refs/tags/0.5.29/crates/uv-trampoline/trampolines/uv-trampoline-aarch64-gui.exe";
+      hash = "sha256-mb8x1FpyH+wy11X5YgWfqh/VUwBb62M4Zf9aFr5V4EE=";
+    };
 
+    aarch64-console = fetchurl {
+      url = "https://raw.githubusercontent.com/astral-sh/uv/refs/tags/0.5.29/crates/uv-trampoline/trampolines/uv-trampoline-aarch64-console.exe";
+      hash = "sha256-1S2aM+6CV7rKz+3ncM5X7kk7uDQeuha1+8lUEMYC75k=";
+    };
+
+    x86_64-gui = fetchurl {
+      url = "https://raw.githubusercontent.com/astral-sh/uv/refs/tags/0.5.29/crates/uv-trampoline/trampolines/uv-trampoline-x86_64-gui.exe";
+      hash = "sha256-icnp1oXrOZpc+dHTGvDbTHjr+D8M0eamvRrC9bPI/KI=";
+    };
+
+    x86_64-console = fetchurl {
+      url = "https://raw.githubusercontent.com/astral-sh/uv/refs/tags/0.5.29/crates/uv-trampoline/trampolines/uv-trampoline-x86_64-console.exe";
+      hash = "sha256-nLopBrlCMMFjkKVRlY7Ke2zFGpQOyF5mSlLs0d7+HRQ=";
+    };
+  };
+in
 buildPythonPackage rec {
   pname = "pex";
-  version = "2.29.0";
+  version = "2.33.1";
   pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-2JeQ1zGKZorVqx+2OzjdF7nln00t42yxP2O0e6AVGx4=";
+    hash = "sha256-kHTvgXe51TencKDkFQAAdyPXuJLBNpJ0NIy1KB8p5JQ=";
   };
 
-  build-system = [ hatchling ];
+  preBuild = ''
+    mkdir -p pex/windows/stubs
+    cp ${uv-trampoline.aarch64-gui} pex/windows/stubs/uv-trampoline-aarch64-gui.exe
+    cp ${uv-trampoline.aarch64-console} pex/windows/stubs/uv-trampoline-aarch64-console.exe
+    cp ${uv-trampoline.x86_64-gui} pex/windows/stubs/uv-trampoline-x86_64-gui.exe
+    cp ${uv-trampoline.x86_64-console} pex/windows/stubs/uv-trampoline-x86_64-console.exe
+  '';
+
+  build-system = [ setuptools ];
 
-  # A few more dependencies I don't want to handle right now...
+  # Module has no tests
```

Comment:
This doesn't seem correct, pex has lots of tests upstream

Reply:
Ah yes, I had assumed they were missing because it was a pypi archive, and removing `doCheck = false` didn't change anything. I've added `pytestCheckHook`, and will submit a new commit when the tests pass.

Reply:
I've gotten the tests up and running, but I suspect they aren't suited to being run on nix; many of them touch the filesystem and fetch from GitHub.
I can continue to ignore swathes of tests, but I fear I'll be excluding a great portion of the test suite.

Reply:
Even with python-xdist, running these tests takes a very long time. At this point, I recommend retaining `doCheck = false`